Unity/opentok/holo/holo/Capturer.mm

```diff
@@ -0,0 +1,373 @@
+    }
+    return self;
+}
+
+-(id)initWithCapturePreset: (NSString*)preset {
+    self = [super init];
+    if (self) {
+        _capturePreset = preset;
+        [[self class] dimensionsForCapturePreset:_capturePreset
+                                           width:&_captureWidth
+                                          height:&_captureHeight];
+        _capture_queue = dispatch_queue_create("com.vonage.holo.Capturer",
+                                               DISPATCH_QUEUE_SERIAL);
+        _videoFrame = [[OTVideoFrame alloc] initWithFormat:
+                       [OTVideoFormat videoFormatNV12WithWidth:_captureWidth
+                                                        height:_captureHeight]];
+    }
+    return self;
+}
+
+- (void)dealloc {
+    [self stopCapture];
+    [self releaseCapture];
+
+    if (_capture_queue) {
+        _capture_queue = nil;
+    }
+    _videoFrame = nil;
+}
+
+- (void)initCapture {
+    _capturer = [[RTC_OBJC_TYPE(VonageRTCCameraVideoCapturer) alloc] init];
+    _depthDataCompressor = [[DepthDataCompressor alloc] init];
+
+    [_capturer setDelegate:self];
+    [_capturer updateDepthDelegate:_depthDataCompressor];
+}
+
+- (void)releaseCapture {
+    [self stopCapture];
+}
+
+- (int32_t)startCapture {
+    _capturing = YES;
+    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, 1 * NSEC_PER_SEC),
+                   dispatch_get_global_queue( DISPATCH_QUEUE_PRIORITY_HIGH, 0), ^{
+        AVCaptureDeviceType deviceType = AVCaptureDeviceTypeBuiltInTrueDepthCamera;
+        AVCaptureDeviceDiscoverySession* deviceDiscoverySession = [AVCaptureDeviceDiscoverySession
+                                                                   discoverySessionWithDeviceTypes:@[ deviceType ]
+                                                                   mediaType:AVMediaTypeVideo
+                                                                   position:AVCaptureDevicePositionFront];
+        AVCaptureDevice* selectedDevice =
+        [deviceDiscoverySession devices]
+        ? [deviceDiscoverySession devices].firstObject
+        : [AVCaptureDevice defaultDeviceWithMediaType:AVMediaTypeVideo];
+
+        AVCaptureDeviceFormat *selectedFormat = nil;
+        int targetWidth = _captureWidth;
+        int targetHeight = _captureHeight;
+        int currentDiff = INT_MAX;
+        NSArray<AVCaptureDeviceFormat *> *formats = [RTC_OBJC_TYPE(VonageRTCCameraVideoCapturer) supportedFormatsForDevice:selectedDevice];
+        for (AVCaptureDeviceFormat *format in formats) {
+            if([format.supportedDepthDataFormats count] == 0){
+                continue;
+            }
+            CMVideoDimensions dimension = CMVideoFormatDescriptionGetDimensions(format.formatDescription);
+            bool isCorrectFps = false;
+            for(AVFrameRateRange* frameRate in format.videoSupportedFrameRateRanges){
+                if(frameRate.maxFrameRate <= 60){
```

Comment:
for high resolution (above 640x480) you we can find formats with 30fps.

Reply:
I did not want to change what you did at https://github.com/Vonage/vonage-media-transformers-samples/blob/feature/OW-744/Unity/ios/VonageUnityIntegration/ViewController.mm#L171. Let me know if you feel we should change this please. Thanks.

Reply:
i think we are good.